core/src/main/java/org/apache/metamodel/CompositeDataContext.java

```diff
@@ -190,9 +183,7 @@ public Schema getSchemaByNameInternal(String name) throws MetaModelException {
                 }
             }
         }
-        String[] result = set.toArray(new String[set.size()]);
-        Arrays.sort(result);
-        return result;
+        return new ArrayList<>(set);
```

Comment:
Ah, missed that. Perfect!

Not sure about List vs. Collection. I think I personally prefer List to avoid wrapping it before working with it, but it's not a huge deal either way.